src/wwwroot/services/auth.js

```diff
@@ -37,25 +37,41 @@
       isUrlAllowed: isUrlAllowed,
       getDefaultUrl: getDefaultUrl
     };
-
+    var loginSession = null;
     var decodedToken = null;
     var encodedToken = null;
     var temporarilyAuthed = false;
 
-    encodedToken = $window.localStorage.getItem('jwtToken');
-    if (encodedToken) {
-      decodeToken();
-      // verify expiration
-    }
+    init();
     return authService;
 
+    function init() {
+      loginSession = getLoginSession('relayLogin');
+      encodedToken = loginSession? loginSession.jwtToken : null;
+      if (encodedToken) {
+        decodeToken();
+        // verify expiration
+      }
+    }
+
+    function getLoginSession(storageName) {
+      var login = angular.fromJson($window.localStorage.getItem(storageName));
+      return login;
+    }
```

Comment:
I am not sure if this function is necessary.

Reply:
I will remove it to call directly angular.fromJson method and covert the local storage.